scripts/table_updates/utilities.py

```diff
@@ -126,12 +186,95 @@ def revert_table_version(syn, table_id):
     """
     table_schema = syn.get(table_id)
     table_columns = syn.getColumns(table_schema.columnIds)
-    table_columns = [col['name'] for col in list(table_columns)]
-    previous_version_num = table_schema.versionNumber-1
-    old_data = syn.tableQuery('SELECT * FROM %s.%s' % (table_id, previous_version_num)).asDataFrame()
+    table_columns = [col["name"] for col in list(table_columns)]
+    previous_version_num = table_schema.versionNumber - 1
+    old_data = syn.tableQuery(
+        "SELECT * FROM %s.%s" % (table_id, previous_version_num)
+    ).asDataFrame()
     old_data = old_data.reset_index(drop=True)
-    table_columns = list(set(table_columns) & set(old_data.columns)) 
+    table_columns = list(set(table_columns) & set(old_data.columns))
     temp_data = old_data[table_columns]
     table_query = syn.tableQuery("SELECT * from %s" % table_id)
     syn.delete(table_query.asRowSet())
-    syn.store(Table(table_schema, temp_data))
\ No newline at end of file
+    syn.store(Table(table_schema, temp_data))
+
+
+def update_tier1a(syn: synapseclient.Synapse, form: str, master_table: pandas.DataFrame, main_genie_table: pandas.DataFrame, column_mapping_table: pandas.DataFrame, bpc_column_list: List[str],logger: logging.Logger = None, cohort: str = "") -> Tuple[str, pandas.DataFrame]:
+    """Replace tier1a variables in patient_characteristics or cancer_panel_test table with Main GENIE release files
+
+    Args:
+        syn (synapseclient.Synapse): The synapse client connection
+        form (str): The form name, can be either patient_characteristics or cancer_panel_test
+        master_table (pandas.DataFrame): Table of all of the primary or irr BPC tables
+        main_genie_table (pandas.DataFrame): The dataframe of Main GENIE release
+        column_mapping_table (pandas.DataFrame): The column mapping table between BPC and Main GENIE
+        bpc_column_list (List[str]): The column list to be replaced
+        logger (logging.Logger, optional): The custom logger. Optional. Defaults to None.
+        cohort (str, optional): The cohort name. Defaults to "".
+
+    Returns:
+        Tuple[str, pandas.DataFrame]: The synapse ID for BPC table to be modified and the updated table as dataframe
+    """
+    # check the validity of bpc_column_list
+    valid_col = column_mapping_table.loc[column_mapping_table["prissmm_form"] == form,].prissmm_element.tolist()
```

Comment:
Why are we using `prissm_form`? Could there be somewhere in the function docstring describing why we are pulling our column list for both bpc and main genie from here?

Reply:
The reason is the prissmm_form matches form column in the [Data Table information table](https://www.synapse.org/Synapse:syn63616766/tables/). I can update the doctring.